### PR TITLE
Wounds UI

### PR DESCRIPTION
**Pull request by Mehuge**
This is a wounds UI based on the following design: https://trello.com/c/30FArgaS/51-wounds-ui

It is a work in progress.  It currently supports the following:
- Blood and Stamina
- PaperDoll (body part only)

The following items are not yet implemented (due to lack of support in the client):
- Other body parts
- Panic
- Temp
- Buffs & Debuffs

The following caveats should be taken into consideration when reviewing this initial pull:-
# Classes / Components

All classes and components are initially being developed locally to the UI.  I expect some of this will eventually be migrated to cu-core and cu-components.
# cu-styles

It does not use cu-styles (I don't know how to use that yet)
# Paper Doll Health

The paper doll health representation for the body part is an emulation because at present the cuAPI does not provide the correct data.  

I calculate the total health of the body part by multiplying the maxHealth value from the injury event by 5 (the total number of health bars you have).  

I calculate the current health value as ((4 - wounds) \* maxHealth) + health where wounds < 5, when wounds is 5, health is set to 0.  That value is used both in the display of current health in the label and in the calculation that works out which level of damage to show.
# SVG

I chose to handle the SVG data internally (coped it into Shapes.tsx) because I need to be able to change the color (and potentially other attributes) in the fly.  This SVG component probably needs expanding and a review to support buffs (which look like they might have borders and stuff).

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/cu-ui/pull/12" target="_blank">ReviewNinja

File: wounds/src/ts/classes/WoundColors.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/// <reference path="../../tsd/tsd.d.ts" />
+
+const defaultColors: string[] = ['#19b24b', '#feeb00', '#fe1e14', '#8a150b', '#200000'];
+
+export class WoundColors {
+	private key: string = "cse.wound.colors";
```
Comment (Lemon-King) [inline wounds/src/ts/classes/WoundColors.tsx]: Lines 12 - 30 do not follow indentation standard of 2 spaces. !fix

File: wounds/src/ts/components/Doll.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    for (let i = 0; i < map.length; i++) {
+      let injury = map.getInjury(i);
+      const part = getPart(i);
+      const maxHealth = (injury.maxHealth * 5);
+      const currentHealth = injury.wounds < 5 ? ((4 - injury.wounds) * injury.maxHealth) + injury.health : 0;
+      const state = getState(currentHealth, maxHealth);
+      const color = this.props.colors.getColorForWound(injury.wounds);
+      parts.push(
+        <Part key={'part.'+part}
+              part={part}
```
Comment (Lemon-King) [inline wounds/src/ts/components/Doll.tsx]: Part has too many space for indentations. !fix

File: wounds/src/ts/components/Doll.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+              health={currentHealth}
+              max={maxHealth}
+              wounds={injury.wounds}
+              state={state}
+              color={color}
+          />
+      );
+      labels.push(
+          <Label key={'part.' + part}
```
Comment (Lemon-King) [inline wounds/src/ts/components/Doll.tsx]: Label has 4 spaces with indentation. !fix

File: wounds/src/ts/components/Part.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export class Part extends React.Component<PartProps, PartState> {
+  render() {
+    const part = this.props.part;
+    const shape : SVGShape = getShape(part, this.props.state);
+    const id: string = 'part' + part;
+    return (
+      <Svg className="part" id={ id } box={ shape.box } color={ this.props.color }
+            polygon={ shape.polygon } path={ shape.path }>
```
Comment (Lemon-King) [inline wounds/src/ts/components/Part.tsx]: Polygon has 6 spaces for indentation instead of 2. !fix

File: wounds/src/ts/components/Svg.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    if (this.props.strokeML) {
+      style += 'stroke-mitrelimit:' + this.props.strokeML + ';';
+    }
+    return (
+      <svg className={this.props.className} viewBox={this.props.box}>
+        <style type="text/css">
+        {'.' + this.props.id + ' {' + style + '}'}
+        </style>
+        <g>
+          {this.props.polygon ?
```
Comment (Lemon-King) [inline wounds/src/ts/components/Svg.tsx:47]: Lines 47 - 64 have too many spaces for indentation. !fix

File: wounds/src/ts/main.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  render() {
+    const character = this.state.character;
+    return (
+      <div>
+        <Doll injuries={character.injuries} colors={this.colors}/>
+        <Buffs type="boon"/>
+        <Buffs type="bane"/>
+        <Stats
+            blood={character.health} bloodMax={character.maxHealth}
```
Comment (Lemon-King) [inline wounds/src/ts/main.tsx]: 48 - 51 have two too many spaces for indentation. !fix

File: wounds/src/ts/classes/WoundColors.tsx (hunk 1 above)
Comment (Lemon-King) [inline wounds/src/ts/classes/WoundColors.tsx]: !resolved

File: wounds/src/ts/components/Doll.tsx (hunk 2 above)
Comment (Lemon-King) [inline wounds/src/ts/components/Doll.tsx]: !resolved

File: wounds/src/ts/components/Doll.tsx (hunk 3 above)
Comment (Lemon-King) [inline wounds/src/ts/components/Doll.tsx]: !resolved

File: wounds/src/ts/components/Part.tsx (hunk 4 above)
Comment (Lemon-King) [inline wounds/src/ts/components/Part.tsx]: !resolved

File: wounds/src/ts/components/Svg.tsx (hunk 5 above)
Comment (Lemon-King) [inline wounds/src/ts/components/Svg.tsx:47]: !resolved

File: wounds/src/ts/main.tsx (hunk 6 above)
Comment (Lemon-King) [inline wounds/src/ts/main.tsx]: !resolved

**Pull request merged**